planner/core/find_best_task.go

```diff
@@ -187,8 +189,9 @@ func (p *LogicalMemTable) findBestTask(prop *property.PhysicalProperty) (t task,
 		return invalidTask, nil
 	}
 	memTable := PhysicalMemTable{
-		DBName: p.dbName,
-		Table:  p.tableInfo,
+		DBName:  p.dbName,
+		Table:   p.tableInfo,
+		Columns: p.tableInfo.Columns,
```

Comment:
Seems we don't need the `Columns` fields because it has been contained in `p.tableInfo`.

Reply:
No, for cluster memory table reader, the column may be pruned in the logical plan.